site-graph/http/handlers.go

```diff
@@ -0,0 +1,78 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"go.uber.org/zap"
+)
+
+type scrapeParams struct {
+	Depth int
+	URL   *url.URL
+}
+
+// parseScrapeParams parse params from client. If depth in query is empty
+// it uses 1 as default parameter. Query parameter url is mandatory and returns
+// error if is not specified.
+func parseScrapeParams(q url.Values) (*scrapeParams, error) {
+	var params scrapeParams
+	if qDepth := q.Get("depth"); qDepth != "" {
+		depth, err := strconv.Atoi(qDepth)
+		if err != nil {
+			return nil, fmt.Errorf("query parameter 'depth' is invalid: %w", err)
+		}
+		params.Depth = depth
+	} else {
+		params.Depth = 1
+	}
+
+	if rawURL := q.Get("url"); rawURL != "" {
```

Comment:
good practice - to put short pieces of code in the beginning. 

E.g. Imagine this if block will be long enough (right now 6 lines but you may add some additional check when code evolves etc. - so it may become 20 lines). When analyzing it  - when I will come  to line 39 you may already not remember what this if condition was about. So you have to return back to line 33 remember and go back to line 39 which is inconvenient

```
if rawurl == "" {
    return  .....
}

// else block - here I can already safely forget what previous if was about
```